.gitignore

```diff
@@ -1,26 +1,68 @@
-# Compiled class file
-*.class
+# Created by .ignore support plugin (hsz.mobi)
 
-# Log file
-*.log
+components/uma-resource-endpoint/target/
+components/uma-resource-service/target/
+target
+# Created by https://www.gitignore.io/api/intellij+all
 
-# BlueJ files
-*.ctxt
+### Intellij+all ###
+# Covers JetBrains IDEs: IntelliJ, RubyMine, PhpStorm, AppCode, PyCharm, CLion, Android Studio and Webstorm
+# Reference: https://intellij-support.jetbrains.com/hc/en-us/articles/206544839
+*.iml
+# User-specific stuff:
+.idea/**/workspace.xml
+.idea/**/tasks.xml
+.idea/dictionaries
+
+# Sensitive or high-churn files:
+.idea/**/dataSources/
+.idea/**/dataSources.ids
+.idea/**/dataSources.xml
+.idea/**/dataSources.local.xml
+.idea/**/sqlDataSources.xml
+.idea/**/dynamic.xml
+.idea/**/uiDesigner.xml
```

Comment:
No need to include each and every file related to .idea. We can completely ignore it.